Sources/SymbolKit/SymbolGraph/Symbol/DeclarationFragments/DeclarationFragments+Simplify.swift

```diff
@@ -0,0 +1,167 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+fileprivate extension SymbolGraph.Symbol.DeclarationFragments.Fragment {
+    init(textFragment text: String) {
+        self.spelling = text
+        self.kind = .text
+        self.preciseIdentifier = nil
+    }
+}
+
+fileprivate extension UnifiedSymbolGraph.Symbol {
+    func declarationFragments(selector: UnifiedSymbolGraph.Selector) -> [SymbolGraph.Symbol.DeclarationFragments.Fragment]? {
+        return (self.mixins[selector]?[SymbolGraph.Symbol.DeclarationFragments.mixinKey] as? SymbolGraph.Symbol.DeclarationFragments)?.declarationFragments
+    }
+
+    func functionSignature(selector: UnifiedSymbolGraph.Selector) -> SymbolGraph.Symbol.FunctionSignature? {
+        return self.mixins[selector]?[SymbolGraph.Symbol.FunctionSignature.mixinKey] as? SymbolGraph.Symbol.FunctionSignature
+    }
+}
+
+internal extension SymbolGraph.Symbol {
+    func overloadSubheadingFragments(
+        argumentSeparator: String = ":",
+        printTrailingArgumentSeparator: Bool = true
+    ) -> [DeclarationFragments.Fragment] {
+        guard let sourceFragments = self.declarationFragments ?? self.names.subHeading ?? self.names.navigator, !sourceFragments.isEmpty else {
+            return [.init(textFragment: self.names.title)]
+        }
+
+        var simplifiedFragments = [DeclarationFragments.Fragment]()
+
+        // In Swift, methods have a keyword as their first token; if the declaration follows that
+        // pattern then pull that out
+        if let firstFragment = sourceFragments.first, firstFragment.kind == .keyword {
+            simplifiedFragments.append(firstFragment)
+        }
+
+        // Then, look for the first identifier, which should contain the symbol's name, and add that
+        if let firstIdentifier = sourceFragments.first(where: { $0.kind == .identifier }) {
+            if !simplifiedFragments.isEmpty {
+                simplifiedFragments.append(.init(textFragment: " "))
+            }
+            simplifiedFragments.append(firstIdentifier)
+        }
+
+        // Assumption: All symbols that can be considered "overloads" are written with method
+        // syntax, including a list of arguments surrounded by parentheses. In Swift symbol graphs,
+        // method parameters are included in the FunctionSignature mixin, so if that's present we
+        // use that to parse the data out.
+
+        simplifiedFragments.append(.init(textFragment: "("))
```

Comment:
The optional initializers make for an interesting situation, since the path component and symbol title don't include the `?` marker. If an optional initializer had the same signature as a non-optional one, the current implementation would combine them. Is this something we want? Do we want to include the marker in the subheading if the title doesn't also include it?